core/vm/interpreter.go

```diff
@@ -265,6 +265,10 @@ func (in *EVMInterpreter) Run(contract *Contract, input []byte, readOnly bool) (
 			in.returnData = res
 		}
 
+		if in.cfg.Debug {
+			in.cfg.Tracer.CaptureStateSpecial(pc, op, gasCopy, cost, callContext, in.returnData, in.evm.depth, err)
```

Comment:
we can do 

`if in.cfg.Debug && op == SSTORE `

so that we can combine `CaptureStateSpecial` and `CaptureState`

Reply:
The `op == SSTORE` is written in the function.